os/pm/pm_procfs.c

```diff
@@ -447,43 +460,117 @@ static size_t power_devices_read(FAR struct file *filep, FAR char *buffer, size_
 }
 
 /****************************************************************************
- * Name: power_lock_write
+* 	Name: power_lock_write
+*   
+* 	Description:
+*	Lock a desired state by increasing the counter of the state.
+*
+* 	Input Parameters:
+*   filep             - File path of the power domain
+*   buffer            - Desired state to be unlocked
+*   buflen (NOT USED) - Length of the buffer
+*
+* 	Returned Value:
+*   0 (OK) means the state lock was successful.
+*	1 (Failed) means the input state was incorrect.
  ****************************************************************************/
 static size_t power_lock_write(FAR struct file *filep, FAR const char *buffer, size_t buflen)
 {
-	int state = atoi(buffer);
-
-	if (state >= PM_NORMAL && state < PM_SLEEP) {
-		pm_stay(0, state);
-		fvdbg("power locked!\n");
-	} else {
-		fdbg("ERROR: Invalid state passed \n");
-	}
+	(void)buffer;
+	(void)buflen;
+
+	/* Always lock PM_NORMAL state, check again whether this implementation
+	is enough to handle, if not, consider below methods...
+	1a. PM_ACTIVITY is invoked in some of the driver, if application utlized it, it should draw state back to PM_NORMAL
+	1b. If 1a cannot cover all condition, add one more syscall api, as a bridge for application to call pm_activity
+	2a. Else, we can consider to lock/unlock for all state other than PM_SLEEP, ie. fetch state: pm_querystate(PM_IDLE_DOMAIN)
+	2b. Need to verify if the transition is going to happen, but lock state appears suddenly, what will happen?
+	*/
+	pm_stay(PM_IDLE_DOMAIN, PM_NORMAL);
+	fvdbg("State locked!\n");
 
-	return buflen;
+	return OK;
 }
 
 /****************************************************************************
- * Name: power_unlock_write
+* 	Name: power_unlock_write
+*   
+* 	Description:
+*	Unlock a desired state by decreasing the counter of the state. A timer 
+*	interval can be provided to setup the timer interrupt. If there is some
+*	other interrupt appear before the interval ends, the system will also
+*	wakeup.
+*
+* 	Input Parameters:
+*   filep          - File path of the power domain
+*   buffer         - Desired state to be unlocked
+*   timer_interval - Timer interval for the timer interrupt
+*
+* 	Returned Value:
+*   0 (OK) means the state unlock was successful.
+*	1 (Failed) means the input state was incorrect.
  ****************************************************************************/
+#ifdef CONFIG_ARCH_CORTEXA32
```

Comment:
Adding architecture specific check in common PM code is not the best method.

Reply:
What is your suggestion on how to add DVFS specific calls to the PM control flow architecture?

Reply:
As mentioned below:

> Hi @edwakuwaku The dvfs implementation in PM looks tightly coupled with tp dual board. I think this is not the right method to implement dvfs. Rather dvfs support should be enabled at board level (arch/arm/src/amebasmart). PM should just call standard dvfs API's exposed by board specific code to tune the frequency. (Similar to how we call up_idle().
> 
> @sunghan-chang What is your opinion ?